### PR TITLE
Merge GA-genotype-dev to GA-main-dev

### PR DESCRIPTION
**Pull request by PiotrLuba**
# What was done?
Added some structural changes to genotype class and const correctness (Adrian).
Added documentation and UTs for all methods (Piotr).
Fixed method GetPairs() (Piotr & Adrian).

### CODE REVIEW

**Review by JohnnyBeet — COMMENTED** [review]
Looks good, approved.

**Review by JohnnyBeet — APPROVED** [review]
Missclick during previous review.

**Review by swirszcz-adrian — APPROVED** [review]

File: SouvlakMVP/SouvlakMVP/Genotype.cs [hunk 1]
```diff
@@ -3,75 +3,97 @@
...
-            (Graph.Vertex, Graph.Vertex) pair = (this[pairID], this[pairID+1]); 
-            output.Add(pair);
+            // Define output list
+            List<(Graph.Vertex, Graph.Vertex)> output = new();
+
+            // Get number of pairs in vertex list, list count is always even
+            int pairsCount = this.Size / 2;
+            for (int pairID = 0; pairID < pairsCount; pairID++)
+            {
+                (Graph.Vertex, Graph.Vertex) pair = (this[2*pairID], this[2*pairID + 1]);
```
Comment (Kri5t0fK) [inline SouvlakMVP/SouvlakMVP/Genotype.cs:84]: Nice, now it works correctly! <3

File: SouvlakMVP/SouvlakMVPTest/TestGenotype.cs
Comment (Kri5t0fK) [inline SouvlakMVP/SouvlakMVPTest/TestGenotype.cs:1]: Looks good. Can't propose more tests to add right now (will add later) ;)

**Review by Kri5t0fK — APPROVED** [review]

**Pull request merged**